gradle/lib/dependabot/gradle/file_parser.rb

```diff
@@ -62,6 +67,79 @@ def self.find_includes(buildfile, dependency_files)
+        parsed_toml_file = parsed_toml_file(toml_file)
+        dependency_set += version_catalog_library_dependencies(parsed_toml_file, toml_file)
+        dependency_set += version_catalog_plugin_dependencies(parsed_toml_file, toml_file)
+        dependency_set
+      end
+
+      # rubocop:disable Metrics/PerceivedComplexity
+      def version_catalog_library_dependencies(parsed_toml_file, buildfile)
+        return DependencySet.new unless parsed_toml_file["libraries"]
+
+        dependency_set = DependencySet.new
+        libraries = parsed_toml_file["libraries"]
+        libraries.each do |_mod, declaration|
+          version = declaration["version"]
+          next if version.nil?
+
+          # Only support basic version and reference formats for now,
+          # refrain from updating anything else as it's likely to be a very deliberate choice.
+          next if (version.is_a?(Hash) && (version.length != 1 || version["ref"].nil?)) || version.include?("[")
+
+          version_details = version["ref"].nil? ? version : "$" + version["ref"]
+          group, name = library_parse_declaration(declaration)
+          next if group.nil? || name.nil?
+
+          details = { group: group, name: name, version: version_details }
+          dependency_set << dependency_from(details_hash: details, buildfile: buildfile)
+        end
+        dependency_set
+      end
+
+      # rubocop:enable Metrics/PerceivedComplexity
+      def library_parse_declaration(declaration)
+        group, name = ""
+        if !declaration["module"].nil?
+          group, name = declaration["module"].split(":")
+        elsif !declaration["group"].nil? && !declaration["name"].nil?
+          group = declaration["group"]
+          name = declaration["name"]
+        else
+          [nil, nil]
+        end
+        [group, name]
+      end
+
+      def version_catalog_plugin_dependencies(parsed_toml_file, buildfile)
+        return DependencySet.new unless parsed_toml_file["plugins"]
+
+        dependency_set = DependencySet.new
+        plugins = parsed_toml_file["plugins"]
+        plugins.each do |_mod, declaration|
+          version = declaration["version"]
+          next if version.nil?
+
+          # Only support basic version and reference formats for now,
+          # refrain from updating anything else as it's likely to be a very deliberate choice.
+          next if (version.is_a?(Hash) && (version.length != 1 || version["ref"].nil?)) || version.include?("[")
+
+          version_details = version["ref"].nil? ? version : "$" + version["ref"]
+          details = { group: "plugins", name: declaration["id"], version: version_details }
+          dependency_set << dependency_from(details_hash: details, buildfile: buildfile)
+        end
+        dependency_set
+      end
+
+      def parsed_toml_file(file)
+        @parsed_file ||= {}
+        @parsed_file[file.name] ||= TomlRB.parse(file.content)
+      rescue TomlRB::ParseError, TomlRB::ValueOverwriteError
```

Comment:
I would think `TomlRB::ValueOverwriteError` this would be an error when writing a Toml file, not when parsing it, but we're rescuing it everywhere together with the `TomlRB::ParseError`, so 👍.